quinn-proto/src/connection/stats.rs

```diff
@@ -0,0 +1,80 @@
+/// Statistics about packets transmitted on a connection
+#[derive(Default)]
+pub struct PacketTransmissionStats {
+    /// The amount of packets transmitted on a connection
+    pub packets: u64,
+    /// The total amount of bytes transmitted on a connection
+    pub bytes: u64,
+}
+
+impl PacketTransmissionStats {
+    /// Calculates the average packet size
+    pub fn avg_packet_size(&self) -> u64 {
+        if self.packets != 0 {
+            self.bytes / self.packets
+        } else {
+            0
+        }
+    }
+}
+
+impl std::fmt::Debug for PacketTransmissionStats {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("PacketTransmissionStats")
+            .field("number of packets", &self.packets)
+            .field("number of bytes", &self.bytes)
+            .field("avg packet size", &self.avg_packet_size())
+            .finish()
+    }
+}
+
+/// Statistics about frames transmitted on a connection
+#[derive(Default)]
+pub struct FrameTransmissionStats {
+    pub acks: u64,
+    pub handshake_done: bool,
+    pub ping: u64,
+    pub path_challenge: u64,
+    pub path_response: u64,
+    pub crypto: u64,
+    pub new_connection_id: u64,
+    pub retire_connection_id: u64,
+    pub datagram: u64,
+    pub reset_stream: u64,
+    pub stop_sending: u64,
+    pub max_stream_data: u64,
+    pub max_data: u64,
+    pub max_streams_uni: u64,
+    pub max_streams_bidi: u64,
+    pub stream: u64,
+}
+
+impl std::fmt::Debug for FrameTransmissionStats {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("FrameTransmissionStats")
+            .field("ACK", &self.acks)
```

Comment:
We can also live with the pure derive. I however kind of liked the frame names, and then started to order alphabetically

Reply:
No objection from me.